packages/Parallel/WorkerRunner.php

```diff
@@ -93,16 +90,23 @@ public function run(Encoder $encoder, Decoder $decoder, Configuration $configura
                     $file = new File($filePath, FileSystem::read($filePath));
                     $this->currentFileProvider->setFile($file);
 
-                    $errorAndFileDiffs = $this->processFiles($file, $configuration, $errorAndFileDiffs);
+                    $fileDiff = $this->chainFileProcessor->process($file, $configuration);
 
-                    if ($errorAndFileDiffs[Bridge::SYSTEM_ERRORS] !== []) {
-                        $this->invalidateFile($file);
-                    } elseif (! $configuration->isDryRun()) {
+                    if ($fileDiff instanceof FileDiff) {
+                        array_unshift($fileDiffs, $fileDiff);
```

Comment:
using unshift here to have current expected e2e test output green.
I'd suggest to move to `fileDiffs[] = fileDiff;` and adapting the expected output files

Reply:
but in another PR